agents/agents-core/src/commonMain/kotlin/ai/koog/agents/core/environment/GenericAgentEnvironment.kt

```diff
@@ -76,7 +77,7 @@ internal class GenericAgentEnvironment(
         toolName: String,
         agentId: String,
         message: String,
-        result: Any?
+        result: JsonObject?
```

Comment:
Is tool result always an object though? E.g. I can return just a `String` or an `Int` from the tool, it won't be an object, right?